pkg/actors/actors.go

```diff
@@ -547,39 +577,45 @@ func (a *actorsRuntime) getOrCreateActor(act *internalv1pb.Actor) *actor {
 	return val.(*actor)
 }
 
-func (a *actorsRuntime) callLocalActor(ctx context.Context, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error) {
-	actorTypeID := req.Actor()
+func (a *actorsRuntime) callLocalActor(ctx context.Context, req *internalv1pb.InternalInvokeRequest) (*internalv1pb.InternalInvokeResponse, error) {
+	act := a.getOrCreateActor(req.GetActor())
 
-	act := a.getOrCreateActor(actorTypeID)
+	// Create the InvokeMethodRequest
+	imReq, err := invokev1.FromInternalInvokeRequest(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create InvokeMethodRequest: %w", err)
+	}
+	defer imReq.Close()
 
 	// Reentrancy to determine how we lock.
 	var reentrancyID *string
 	if a.actorsConfig.GetReentrancyForType(act.actorType).Enabled {
-		if headerValue, ok := req.Metadata()["Dapr-Reentrancy-Id"]; ok {
-			reentrancyID = &headerValue.GetValues()[0]
+		if md := imReq.Metadata()["Dapr-Reentrancy-Id"]; md != nil && len(md.GetValues()) > 0 {
+			reentrancyID = ptr.Of(md.GetValues()[0])
```

Comment:
Please add an example about whole value.

Reply:
I'm not sure I follow your ask. This is not new code, it was there before